x/lockup/types/msgs_test.go

```diff
@@ -14,91 +14,246 @@ import (
+			} else {
+				require.Error(t, test.msg.ValidateBasic(), "test: %v", test.name)
+			}
+		})
+		
+	}
+}
+
+func TestMsgBeginUnlocking(t *testing.T) {
+	appParams.SetAddressPrefixes()
+	addr1, invalidAddr := generateTestAddrs()
 
 	tests := []struct {
 		name       string
-		msg        MsgLockTokens
+		msg        MsgBeginUnlocking
 		expectPass bool
 	}{
 		{
 			name: "proper msg",
-			msg: createMsg(func(msg MsgLockTokens) MsgLockTokens {
-				// Do nothing
-				return msg
-			}),
+			msg: MsgBeginUnlocking{
+				Owner: addr1,
+				ID:    1,
+				Coins: sdk.NewCoins(sdk.NewCoin("test", sdk.NewInt(100))),
+			},
 			expectPass: true,
 		},
 		{
 			name: "invalid owner",
-			msg: createMsg(func(msg MsgLockTokens) MsgLockTokens {
-				msg.Owner = invalidAddr.String()
-				return msg
-			}),
-			expectPass: false,
+			msg: MsgBeginUnlocking{
+				Owner: invalidAddr,
+				ID:    1,
+				Coins: sdk.NewCoins(sdk.NewCoin("test", sdk.NewInt(100))),
+			},
 		},
 		{
-			name: "invalid duration",
-			msg: createMsg(func(msg MsgLockTokens) MsgLockTokens {
-				msg.Duration = -1
-				return msg
-			}),
-			expectPass: false,
+			name: "invalid lockup ID",
+			msg: MsgBeginUnlocking{
+				Owner: addr1,
+				ID:    0,
+				Coins: sdk.NewCoins(sdk.NewCoin("test", sdk.NewInt(100))),
+			},
 		},
 		{
-			name: "invalid coin length",
-			msg: createMsg(func(msg MsgLockTokens) MsgLockTokens {
-				msg.Coins = sdk.NewCoins(sdk.NewCoin("test1", sdk.NewInt(100000)), sdk.NewCoin("test2", sdk.NewInt(100000)))
-				return msg
-			}),
-			expectPass: false,
+			name: "invalid coins length",
+			msg: MsgBeginUnlocking{
+				Owner: addr1,
+				ID:    1,
+				Coins: sdk.NewCoins(sdk.NewCoin("test1", sdk.NewInt(100000)), sdk.NewCoin("test2", sdk.NewInt(100000))),
+			},
 		},
 		{
-			name: "zero token amount",
-			msg: createMsg(func(msg MsgLockTokens) MsgLockTokens {
-				msg.Coins = sdk.NewCoins(sdk.NewCoin("test1", sdk.NewInt(0)))
-				return msg
-			}),
-			expectPass: false,
+			name: "not positive coins amount",
+			msg: MsgBeginUnlocking{
+				Owner: addr1,
+				ID:    1,
+				Coins: sdk.NewCoins(sdk.NewCoin("test1", sdk.NewInt(0))),
+			},
 		},
 	}
 
 	for _, test := range tests {
-		if test.expectPass {
-			require.NoError(t, test.msg.ValidateBasic(), "test: %v", test.name)
-		} else {
-			require.Error(t, test.msg.ValidateBasic(), "test: %v", test.name)
-		}
+		t.Run(test.name, func(t *testing.T) {
+			if test.expectPass {
+				require.NoError(t, test.msg.ValidateBasic(), "test: %v", test.name)
+				require.Equal(t, test.msg.Route(), RouterKey)
+				require.Equal(t, test.msg.Type(), "begin_unlocking")
+				signers := test.msg.GetSigners()
+				require.Equal(t, len(signers), 1)
+				require.Equal(t, signers[0].String(), addr1)
+			} else {
+				require.Error(t, test.msg.ValidateBasic(), "test: %v", test.name)
+			}
+		})
+		
 	}
 }
 
-// TODO: Complete table driven tests for the remaining messages
+func TestMsgExtendLockup(t *testing.T) {
```

Comment:
Does it make sense to define these tests on the test suite?

Reply:
Do u mean to use the test suite instead of testing.T?

Reply:
Sorry for not being clear enough.

What I meant is to define something like this:
https://github.com/osmosis-labs/osmosis/blob/fe98f6e4453cd32035d277c21ef2f3669b677bb2/x/lockup/keeper/keeper_test.go#L15

Instead, we would have `TypesTestSuite`

Reply:
I think we should create an issue for that. I just took a look through all modules and tests in `types` all use testing.T
Should refactor all those tests in a pr right?

Reply:
If that's the case, let's keep this as is. I think consistency is more important here. Thanks for double-checking that